loaders.py

```diff
@@ -68,6 +70,70 @@ def __getitem__(self, index):
             return Xi
 
 
+class MetadataImageSegmentationDatasetDistances(Dataset):
+    def __init__(self, X, y, train_mode,
+                 image_transform, image_augment_with_target,
+                 mask_transform, image_augment,
+                 distance_matrix_transform):
+        super().__init__()
+        self.X = X
+        if y is not None:
+            self.y = y
+        else:
+            self.y = None
+
+        self.train_mode = train_mode
+        self.image_transform = image_transform
+        self.mask_transform = mask_transform
+        self.distance_matrix_transform = distance_matrix_transform
+        self.image_augment = image_augment
+        self.image_augment_with_target = image_augment_with_target
+
+    def load_image(self, img_filepath):
+        image = Image.open(img_filepath, 'r')
+        return image.convert('RGB')
+
+    def load_distances(selfself, dist_filepath):
+        return joblib.load(dist_filepath)
+
+    def __len__(self):
+        return self.X.shape[0]
+
+    def __getitem__(self, index):
+        img_filepath = self.X[index]
+        Xi = self.load_image(img_filepath)
+
+        if self.y is not None:
+            mask_filepath = self.y[index]
+            Mi = self.load_image(mask_filepath)
+            distance_filepath = mask_filepath.replace("/masks/", "/distances/")[:-4]
```

Comment:
@apyskir I presume this action is performed to drop `.jpg` .In the future we may have a competition with `.jpeg` images or something. I think you can call `from os.path import basename; basename(filename)`

Reply:
@jakubczakon `os.path.splitext(filename)[0]` is what actually does the job